xml/readme.tex

```diff
@@ -247,6 +251,62 @@ \subsection{Example of a \tag{types} tag}
 typedef ptrdiff_t GLintptr;
 \end{verbatim}
 
+\section{Kinds (\tag{kinds} tag)}
+\label{tag:kinds}
+
+The \tag{kinds} tag contain individual \tag{kind} tags describing
+some metadata not captured by the \tag{types} tag (see section~\ref{tag:types}) 
+or \tag{enums} tag (see section~\ref{tag:groups}).
+
+\subsection{Attributes of (\tag{kinds} tag)}
+
+\begin{itemize}
+\item \attr{name} - the name of the kind.
+\item \attr{desc} - a description of what the kind indicates.
+\end{itemize}
+
+\subsection{Contents of \tag{kinds} tags}
+
+Each \tag{kinds} block contains zero or more \tag{kind} tags, in
+arbitrary order.
+
+\subsection{Example of a \tag{kinds} tag}
+
+\begin{verbatim}
+<kinds>
+    <kind name="Color" desc="This parameter represents part of or a complete color." />
+    <kind name="Clamped[0; 1]" desc="This parameter will get clamped to the 0 to 1 range." />
+</kinds>
+\end{verbatim}
+
+\section{Kind (\tag{kind} tag)}
+\label{tag:kind}
+
+Each \{kind} tag defines a single kind annotation.
+
+\subsection{Attributes of \tag{kind} tag}
+
+\begin{itemize}
+\item \attr{name} - kind name, an arbitrary string used
+      to identify this kind.
+\item \attr{desc} - kind description, a description of what it
```

Comment:
Since ',' is the separator for multiple kind annotations, the kind 'desc' presumably cannot contain ',' - this should be mentioned here.

Reply:
`desc` is a description of one kind, it can contain anything. Did you mean the `name` attribute?

Reply:
I think I have resolved this now.